### ISSUE TITLE
Eye contact doesn't keep looking at camera, frequently looks away

### ISSUE THREAD

**Issue by kurenai-11** [alloystorm/dvvr#381]
**Describe the bug**
When you enable eye contact, every couple of seconds the character loses focus and looks away (where nothing exists) for no apparent reason, even when the camera itself doesn't move.

**To Reproduce**
1. Enable "eye contact" in settings - "lifelike motions" tab.
2. Observe the problem. Exists with any character and every setting I could think of trying, both in Desktop and VR.

**Expected behavior**
The character should be able to keep eye contact with camera without losing focus and looking away at ghosts or something every couple of seconds.

**Video**

https://github.com/user-attachments/assets/003fc2be-e45a-4b52-b183-0270c089713c

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Game version: 2024.11.1440

**Comment by alloystorm** [comment]
Yes, that's intentionally done to mimic human behavior. 

Staring at people constantly is actually creepy.

**Comment by kurenai-11** [comment]
But maybe it's better to have some customization then? Looking away every couple of seconds feels a little bit too frequent, especially somewhere where nothing exists at all. Besides, some loopable motions are probably better with constant eye contact.

Turning off "Random Target" doesn't change anything, nor any other setting I tried.

**Comment by alloystorm** [comment]
You can also set "Look At Peers" and "Look At Body" to 0 and that should make them focus more on the main camera. 

But they will still look away from time to time, that is programmed purposefully to be less robotic... If there's no target in view they'll just look stright ahead. 

We'll consider adding a "stare" option in the future.

**Comment by alloystorm** [comment]
A new "Stare" mode is added in 2025.1, which will force the actor to look at the closest target constantly.

**Comment by username-011** [comment]
@alloystorm this actually still isn't fixed. Characters still look away. With the new "stare" option, they look away less than without it, but they still can't keep looking at the camera 100% of the time.

I think it is easy to confirm by itself, if you enable the new "stare" version, it doesn't work as expected. Every 10-15 seconds or so, the character looks away for a second, then returns.
Here is a video demonstration:

https://github.com/user-attachments/assets/156ad180-9623-4f58-8481-7020b73c9fb3

And the head moves too:

https://github.com/user-attachments/assets/397927e4-5562-49bb-a959-783b6092893a

**Comment by alloystorm** [comment]
Most likely they "disconnect" briefly when they blink, which is actually by design because the assumption is that when eyes are closed they can't see therefore no eye contact possible. You can try to turn off blink and see if that does what you wanted.

We might revise that logic in the future.

**Comment by username-011** [comment]
@alloystorm 
> You can try to turn off blink and see if that does what you wanted.

No, unfortunately it doesn't change anything. The character still looks away. Here is another example: 

https://github.com/user-attachments/assets/d0376e7a-0bd8-4ec2-9c2c-35a5c4233284

https://github.com/user-attachments/assets/79ac3a2e-02bd-49b0-966e-dce985936ea9

You can check it for yourself to confirm that it works like that currently. I did it on the version 2026.3.1833, but I think every version has this problem.

**Comment by Chrisc78** [comment]
> [@alloystorm](https://github.com/alloystorm)
> 
> > You can try to turn off blink and see if that does what you wanted.
> 
> No, unfortunately it doesn't change anything. The character still looks away. Here is another example:
> 
>  2026-04-22.09-35-49.mp4 
>  2026-04-22.09-37-26.mp4 
> You can check it for yourself to confirm that it works like that currently. I did it on the version 2026.3.1833, but I think every version has this problem.

Are those FBX models? if so connect the bones "Torso2 to Chest"

**Comment by username-011** [comment]
> Are those FBX models? if so connect the bones "Torso2 to Chest"

no, those aren't FBX. Those are just normal PMX. And I think this bug exists on every single model I have.
Do you also have this bug by the way, or your characters don't look away ever? Can you test?

**Comment by Chrisc78** [comment]
I will look

**Comment by Chrisc78** [comment]
it never bothered me but every few seconds her eyes look away. I tried to see if I could stop it without luck

**Comment by username-011** [comment]
yeah it's probably indeed a confirmed bug then

**Comment by alloystorm** [comment]
Fixed in 2026.5

**Comment by Chrisc78** [comment]
> yeah it's probably indeed a confirmed bug then

<img width="1580" height="1330" alt="Image" src="https://github.com/user-attachments/assets/3457b919-6f64-4ef8-9a5a-466bd49cc500" />

Your right this is so much better in 2026.5, I find this to be my favorite settings. it depends on the model if "Cartoon Eyes" needs to be turned Off or On, I find Eye contact Angle and Head Turn settings should be maxed

**Comment by username-011** [comment]
For me if I make Eye Contact Angle more than 90, it breaks for some models from some angles (the iris goes out of the socket or the character looks down where no one is). While less than or equal 90 works as expected.

**Comment by Chrisc78** [comment]
I found that some models you have to turn off Cartoon Eyes for it to work correctly.

**Comment by username-011** [comment]
> I found that some models you have to turn off Cartoon Eyes for it to work correctly.

Most of my models are converted models from other formats (most from Koikatsu). By default, they don't work without Cartoon Eyes. Only rarely I may also added morph based eye control there instead of Cartoon Eyes, but it needs editing of the `.pmx` for that. I used it only when Cartoon eyes didn't work. I didn't test morph based eye control with the new version yet though, don't know if it works better or also broken. I'll test some time.

**Comment by username-011** [comment]
Also in the latest version some models for which the shape eye morph eye control previously worked, now don't work properly. In 2026.3 the character looks at the camera properly, while in 2026.5, they are broken (the character look somewhere else). I may do a demonstration some time and maybe create a different issue then
